src/backend/aspen/app/views/phylo_runs.py

```diff
@@ -0,0 +1,188 @@
+        joinedload(Sample.uploaded_pathogen_genome, innerjoin=True),
+    )
+    # Step 3 - Enforce AuthZ
+    all_samples = authz_sample_filters(all_samples, sample_ids, user)
+
+    pathogen_genomes: MutableSequence[PathogenGenome] = list()
+    missing_sample_ids = list(sample_ids)  # Make a copy of sample id's
+    for sample in all_samples:
+        pathogen_genomes.append(sample.uploaded_pathogen_genome)
+        try:
+            missing_sample_ids.remove(sample.private_identifier)
+        except ValueError:
+            pass
+        try:
+            missing_sample_ids.remove(sample.public_identifier)
+        except ValueError:
+            pass
+    if len(pathogen_genomes) == 0:
+        raise ValueError("No sequences selected for run.")
+
+    if missing_sample_ids:
+        sentry_sdk.capture_message(
+            f"User requested sample id's they didn't have access to ({missing_sample_ids})"
+        )
+
+    aligned_gisaid_dump = (
+        g.db_session.query(AlignedGisaidDump)
+        .join(AlignedGisaidDump.producing_workflow)
+        .order_by(Workflow.end_datetime.desc())
+        .first()
+    )
+    if not aligned_gisaid_dump:
+        return Response("No gisaid dump for run", 500)
+
+    template_path_prefix = (
+        "/usr/src/app/aspen/workflows/nextstrain_run/builds_templates"
+    )
+    builds_template_file = (
+        f"{template_path_prefix}/{PHYLO_TREE_TYPES[request_data['tree_type']]}"
+    )
+    builds_template_args = {
+        "division": group.division,
+        "location": group.location,
+    }
+    start_datetime = datetime.datetime.now()
+
+    workflow: PhyloRun = PhyloRun(
+        start_datetime=start_datetime,
+        workflow_status=WorkflowStatusType.STARTED,
+        software_versions={},
+        group=group,
+        template_file_path=builds_template_file,
+        template_args=builds_template_args,
+        name=request_data["name"],
+    )
+
+    workflow.inputs = list(pathogen_genomes)
+    workflow.inputs.append(aligned_gisaid_dump)
+
+    g.db_session.add(workflow)
+    g.db_session.flush()
+
+    responseschema = PhyloRunResponseSchema()
+
+    # TODO - invoke a step function!
+    # boto3.invok_step_something(parameters)
+    aspen_config = application.aspen_config
+
+    sfn_input_json = {
+        "Input": {
+            "Run": {
+                "aspen_config_secret_name": os.environ.get(
+                    "ASPEN_CONFIG_SECRET_NAME", "aspen-config"
+                ),
+                "aws_region": aws.region(),
+                "docker_image_id": aspen_config.NEXTSTRAIN_DOCKER_IMAGE_ID,
+                "group_name": group.name,
+                "remote_dev_prefix": os.getenv("REMOTE_DEV_PREFIX"),
+                "s3_filestem": f"{group.location}/{request_data['tree_type'].capitalize()}",
+                "workflow_id": workflow.id,
+            },
+        },
+        "OutputPrefix": aspen_config.NEXTSTRAIN_OUTPUT_PREFIX,
```

Comment:
TODO -- we need to specify a different output prefix for every SFN call. We should update this to include the workflow ID.

Reply:
updated!